modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParser.java

```diff
@@ -147,12 +155,64 @@ private void parseStructure(ExtensionElement extensionElement) {
 
   private Set<String> parseSupportedJavaVersions(ExtensionElement extensionElement) {
     return extensionElement.getValueFromAnnotation(JavaVersionSupport.class)
-        .map(a -> a.getEnumArrayValue(JavaVersionSupport::value).stream()
+        .map(a -> a
+            .getEnumArrayValue(javaVersionSupport -> getJavaVersionSupport(javaVersionSupport, extensionElement))
+            .stream()
             .map(JavaVersion::version)
             .collect(toCollection(() -> (Set<String>) new LinkedHashSet<String>())))
         .orElse(emptySet());
   }
 
+  private JavaVersion[] getJavaVersionSupport(JavaVersionSupport javaVersionSupport, ExtensionElement extensionElement) {
+    try {
+      return javaVersionSupport.value();
+    } catch (EnumConstantNotPresentException e) {
+      // An enum value not present in the Mule container version of `JavaVersion` is in use, we need to introspect the annotation
+    }
+
+    try {
+      Class<?> annotatedClass = extensionElement
+          .getDeclaringClass()
+          .orElseThrow(() -> new RuntimeException(format("Expected extension element '%s' to have a declaring class",
```

Comment:
rethrow the original EnumConstantNotPresentException with the `NoSuchElementException` as suppressed.

Reply:
Done.